five-minute-serverless-backend/README.md

```diff
@@ -0,0 +1,209 @@
++  };
++;
++
+ exports.handler = async (body, context) => {
+```
+
+This function uses the execution `context` to create a custom `HTTPResponse` object and adds the relevant headers to the response. You can use this object to customize response headers, the response body and the HTTP status code. Please note that when using custom responses, Binaris no longer automatically JSON encodes the output so you need to do this yourself.
+
+Now all we need to do is wrap our handler with this functions and we're done:
+
+```diff
+-exports.handler = async (body, context) => {
++exports.handler = CORS(async (body, context) => {
+   return (typeof(body) === 'string' ? body : '').toUpperCase();
+-};
++});
+```
+
+Our logic remains unchanged. We can now re-deploy and test:
+
+```
+$ bn delpoy public_backend
+$ curl -i https://run.binaris.com/v2/run/<account id>/public_backend -H 'Content-Type: application/json' -d '"hello"'
+HTTP/1.1 200 OK
+Server: openresty/1.13.6.1
+Date: Thu, 14 Feb 2019 23:16:06 GMT
+Content-Type: application/json; charset=utf-8
+Content-Length: 7
+Connection: keep-alive
+X-Binaris-Bolt-Duration-Usecs: 1190.336
+Access-Control-Allow-Origin: *
+Access-Control-Allow-Headers: Origin, X-Requested-With, Content-Type, Accept
+X-Binaris-Request-ID: 89429871a0c1c8f5d9af68ab227bf9e3
+
+"HELLO"
+```
+
+Notice the ``Access-Control-Allow-*`` headers in the response.
+
+Here is the final code for our backend `function.js`:
+
+```javascript
+const CORS = (handler) =>
+  async (body, context) => {
+    const response = {
+      statusCode: 200,
+      headers: {
+        'Access-Control-Allow-Origin': '*',
+        'Access-Control-Allow-Headers': 'Origin, X-Requested-With, Content-Type, Accept',
+      },
+    };
+    if (context.request.method !== 'OPTIONS') {
+      const output = await handler(body);
+      response.headers['Content-Type'] = 'application/json';
+      response.body = JSON.stringify(output);
+    }
+    return new context.HTTPResponse(response);
+  };
+;
+
+exports.handler = CORS(async (body, context) => {
+  return (typeof(body) === 'string' ? body : '').toUpperCase();
+});
+```
+
+## Call your function from frontend code
+
+Our backend is now live and we can use a simple HTML app to call it. The app code is in a single file named `frontend.html`.
+
+First you need to plug in your Binaris account id. At the very top of the file, locate the following line:
+
+```htmlmixed
+    <script>const BINARIS_ACCOUNT_ID = '<account id>'</script>
+```
+
+Replace `<account id>` with your actual account id.
+
+You can now open `frontend.html` in your browser to run the app. Simply type in some text and hit Enter. The text will be sent to the backend and returned in upper-case to be displayed on screen.
+
+Calling our backend is a simple HTTP POST request. We included our own wrapper for `XMLHttpRequest` mostly to handle HTTP request and response headers. Our wrapper, called `request()` also returns a Promise so that the code calling it is cleaner.
```

Comment:
Unless this is urgent, you can wait until the SDK will be released and avoid all this.
also, `request` in this context might not be the best name sine the interface differs from the very very common `request` package and the source for the wrapper is not here (you'll have to look in the html source itself to see it)
If you don't want to wait for the SDK, I recommend using fetch (https://fetch.spec.whatwg.org/#dom-global-fetch). It will remove the need for wrapper

Reply:
Renamed request() to sendHttpRequest(). If the SDK comes out before this is published we should change